azure/index.d.ts

```diff
@@ -377,18 +362,19 @@ export type AzExtGenericClientInfo = AzExtGenericCredentials | { credentials: Az
  * 2. Uses resourceManagerEndpointUrl to support sovereigns (if clientInfo corresponds to an Azure environment)
  * @param clientInfo The client/credentials info or `undefined` if no credentials are needed
  */
-export function createGenericClient(context: IActionContext, clientInfo: AzExtGenericClientInfo): Promise<ServiceClient>;
+export declare function createGenericClient(context: IActionContext, clientInfo: AzExtGenericClientInfo | undefined): Promise<ServiceClient>;
```

Comment:
```suggestion
export declare function createGenericClient(context: IActionContext, clientInfo?: AzExtGenericClientInfo): Promise<ServiceClient>;
```

Reply:
I think the idea for it to be undefined was so that you could create a generic client to make a one-off request to an endpoint without having to specify a client sdk info.

Reply:
Yeah but isn't my suggestion just changing the syntax and not the result?

Reply:
Oh yeah, I didn't see the ?.

Which leads me to why I'd prefer not to do your suggestion. When you mark it as optional, it makes omitting `clientInfo` less intentional. 

Whereas with the old syntax, I believe that you _must_ type `undefined` in the parameter list. It kind of enforces whoever is using the command to acknowledge that they are not passing in anything on purpose.

Reply:
Nice, I guess that's a perfect scenario to not use `?`. 👍